### ISSUE TITLE
Incompatible bleedingEdge Connection stub typehint with DBAL 4

### ISSUE THREAD

**Issue by trearcul** [phpstan/phpstan-doctrine#559]
Hi,

I hope I'm not creating duplicity here, but I've just run into an incompatibility with the bleeding-edge stub of `DBAL\Connection` added in https://github.com/phpstan/phpstan-doctrine/pull/546. This issue arises with DBAL 4, leading to false positives when an array of `ParameterType|ArrayParameterType` enums is passed as the `$types` argument of `Connection::execute*` methods.

E.g
```php
namespace Vendor\App;

use Doctrine\DBAL\Connection;
use Doctrine\DBAL\ParameterType;

final readonly class DbalFooFinder
{
    public function __construct(private Connection $connection)
    {
    }

    /** @return list<array{bar: string, baz: string}> */
    public function find(string $baz): array
    {
        /** @var list<array{bar: string, baz: string}> $result */
        $result = $this->connection
            ->executeQuery(
                'SELECT bar, baz FROM foo WHERE baz = :baz',
                ['baz' => $baz],
                ['baz' => ParameterType::STRING],
            )
            ->fetchAllAssociative();

        return $result;
    }
}
```

Results in

```
 ------ --------------------------------------------------------------------------------
  Line   src/DbalFooFinder.php
 ------ --------------------------------------------------------------------------------
  23     Parameter #3 $types of method Doctrine\DBAL\Connection::executeQuery() expects
         array<int|string, Doctrine\DBAL\Types\Type|int|string|null>, array<string,
         Doctrine\DBAL\ParameterType::STRING> given.
 ------ --------------------------------------------------------------------------------
```

Even though the DBAL 4's typehint for `$types` parameter is
```
/**
 * @psalm-type WrapperParameterType = string|Type|ParameterType|ArrayParameterType
 * @psalm-type WrapperParameterTypeArray = array<int<0, max>, WrapperParameterType>|array<string, WrapperParameterType>
 */

/**
 * @psalm-param WrapperParameterTypeArray $types
 */
```

**Comment by VincentLanglet** [comment]
Hi @trearcul, can you try https://github.com/phpstan/phpstan-doctrine/pull/561 ?

**Comment by VincentLanglet** [comment]
HI @trearcul, can you try #561 ?

**Comment by trearcul** [comment]
Hi @VincentLanglet, awesome, your fix works with ParameterType enums like charm. Thanks for fast response from both of you! ❤️

**Comment by ondrejmirtes** [comment]
Fixed by https://github.com/phpstan/phpstan-doctrine/pull/561 and released as 1.3.67.
